quasselgrep/db.py

```diff
@@ -14,6 +14,13 @@ def connect(self, options):
 
 			self.connection = dbmodule.connect(options.db_name, check_same_thread=False)
 			cursor = self.connection.cursor()
+
+			#Newer sqlite versions have timestamps in milliseconds.
+			cursor.execute('SELECT value FROM coreinfo WHERE key="schemaversion"')
+			results = cursor.fetchall()
+			if len(results) != 1:
+				raise ValueError('Incorrect sqlite schemaversion format')
+			options.sqlite_version = results[0][0]
```

Comment:
This results in comparing a `str` with an `int` later, for example…

```
  File "$HOME/Desktop/fdsa/quasselgrep-sqlite3_ts_ms/quasselgrep/query.py", line 145, in columns
    if self.options.sqlite_version >= 31:
TypeError: '>=' not supported between instances of 'str' and 'int'
```

**NOTE:** I would suggest providing some form of fallback or catch in case the results cannot be casted to an integer.  Quassel should keep schema versions as whole numbers, but guarding against future changes with an informative error is probably a good idea.

```suggestion
			try:
				# Schema version should be an integer, but this isn't guaranteed
				options.sqlite_version = int(results[0][0])
			except ValueError as ex:
				raise ValueError('Unexpected sqlite schemaversion "{0}", not an integer'.format(results[0][0])) from ex
```

With this fix, your changes work on an old and new SQLite database!

### Hypothetical robustness test
Replacing the result with the string `test-unexpected` confirms that this error handling works.

```
Error connecting to database: Unexpected sqlite schemaversion "test-unexpected", not an integer
```

Reply:
Thanks for testing! I run quasselgrep with python2 (because this is what the shebang points to) where the crash doesn't happen. Changing the shebang to python3 doesn't work on my distro (pkg_resources.DistributionNotFound).

Anyhow, the "from ex" syntax isn't python2 compatible. Therefore I've slightly modified your patch by dropping "from ex" and adapting your ValueError string to match the global coding style (e instead of ex, '%' instead of format()). I've also tried to replace the result with the string '1.0.x' and it works:
`Error connecting to database: Unexpected sqlite schemaversion 1.0.x, not an integer: invalid literal for int() with base 10: '1.0.x'`

Perhaps you can retest with python3 to make sure I didn't break anything?

Reply:
I missed this reply until now - that makes sense!  I overlooked Python 2 support, thank you for catching that, too.

As retested earlier, this looks good!